bids/layout/layout.py

```diff
@@ -587,6 +619,10 @@ def check_for_description(dir):
             # Default config and sources values
             kwargs['config'] = kwargs.get('config') or ['bids', 'derivatives']
             kwargs['sources'] = kwargs.get('sources') or self
+            if create_derivative_database_files:
```

Comment:
One concern about this approach, if I understand it correctly, is that if the derivative DB file has previously been created, but the derivative dataset is updated with new files etc., passing a `database_file` to the root `BIDSLayout` will cause the derivative layout to use the existing DB file even if the `database_file` argument is different. Is this right?

If so, I think we may need to rework this slightly—e.g., by using the root `database_file` as part of the derivative `database_file` name. (I realize that the user can explicitly pass `reset_database=True` to address thihs, but if they're passing an entirely new `database_file` name to the root, they would have no reason to expect the newness to fail to propagate, since they probably have no idea what DB name the derivatives are internally using).

Reply:
I was thinking that a) you could delete the .sql files, or b) you could use 'reset_database=True'

To address your concern I have added:
```python
reset_database=index_dataset or reset_database,

```
```python
        self.add_derivatives(
                derivatives, create_derivative_database_files=create_derivative_database_files,
                validate=validate, absolute_paths=absolute_paths,
                derivatives=None, config=None, sources=self, ignore=ignore,
                force_index=force_index, config_filename=config_filename,
                regex_search=regex_search, reset_database=index_dataset or reset_database,
                index_metadata=index_metadata)
```